### DISCUSSION TITLE
How to add an external library.

### DISCUSSION THREAD

**Discussion by nprange-philips** [ajaxorg/ace#4940] [Q&A] [answered]
Hello,

I'm looking for a way to add an external library to the project so I can connect to a database.
When I try to `var oracledb = require('oracledb');` the following error is displayed:

`GET http://localhost:8888/oracledb.js 404 (Not Found)                  require.js:495`
`Error loading oracledb required from ace/ext/language_tools       require.js:134`

Has anyone run into this problem before?

**Comment by andrewnester** [comment, marked as answer]
It does not relate to Ace, you need to use some sort of bundlers, for example Webpack
